### PR TITLE
Add Edit and Del

### PR DESCRIPTION
**Pull request by M1rn4**
In this pull request, we create a complete webpage  book with classes of JavaScript

# Using Webpack

* Create a new repository
* Add modules
* Refactored index.js
* Used luxon library
* Used linters
:100:

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -19,41 +18,98 @@ const showList = (array) => {
...
+  const ele = tasks.findIndex((e) => e.index + 1 === taskItem);
+  const taskName = document.getElementById(taskItem);
+  if (taskName.value === '') {
+    // eslint-disable-next-line no-alert
+    window.alert('can not save, empty value');
+    return;
+  }
+  tasks[ele].description = taskName.value;
+  localStorage.setItem('tasks', JSON.stringify(tasks));
 };
```
Comment (Kamba56) [inline src/index.js]: - [ ] Please add a new js file to implement the new functionalities added and import them as a module. This is part of the project requirements.

File: src/index.js [hunk 2]
```diff
@@ -19,41 +18,98 @@ const showList = (array) => {
...
+  taskItem = parseInt(taskItem, 10);
+  const ele = tasks.findIndex((e) => e.index + 1 === taskItem);
+  const taskName = document.getElementById(taskItem);
+  if (taskName.value === '') {
+    // eslint-disable-next-line no-alert
+    window.alert('can not save, empty value');
+    return;
+  }
+  tasks[ele].description = taskName.value;
+  localStorage.setItem('tasks', JSON.stringify(tasks));
```
Comment (Kamba56) [inline src/index.js]: - [ ] I noticed that after editing a task, the edited task reverted back to its original form when I refreshed the browser. Kindly ensure the edited task persists by correctly storing it in the database. You can implement this on change of the input value.

File: src/index.js [hunk 3]
```diff
@@ -19,41 +18,98 @@ const showList = (array) => {
...
+};
+
+const deleteOne = (array) => {
+  let taskName = array.children[0].children[0].children[1].id;
+  taskName = parseInt(taskName, 10);
+  const tasks = JSON.parse(localStorage.getItem('tasks'));
+  const ele = tasks.findIndex((e) => e.index === taskName);
+  tasks.splice(ele - 1, 1);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
```
Comment (Kamba56) [inline src/index.js]: - [ ] After removing a task, kindly update the indexes of the remaining tasks so they represent the current list order and are unique. Example
```javascript
tasks = [
  { 
     description: task1,
    index: 1
  },
  { 
     description: task2,
    index: 2
  },
  { 
     description: task3,
    index: 3
  }
]
```
If the task with index: 2 is removed. The indexes should be updated like;
```javascript
tasks = [
  { 
     description: task1,
    index: 1
  },
  { 
     description: task3,
    index: 2
  }
]
```

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @M1rn4 , 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great use of Github workflow!👏👏👏
- [x] All linters passed

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (M1rn4) [reply src/index.js]: you need to click on the save image to save the edited changes.

**Review by Kamba56 — APPROVED** [review]
Hi @M1rn4 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
## Highlights
- [x] All linters passed!👏👏👏
- [x] Great use of Github workflow!👏👏👏
- [x] Descriptive PR!👏👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**